# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/base/BodyConsumer.cpp
# Commit: ac548c2a725b
# Full Hash: ac548c2a725b66db57c81bb24accca2bbe075931
# Author: Olli Pettay <Olli.Pettay@helsinki.fi>
# Date: 2019-07-29 21:38:39
# Description:
#   Bug 1565930, try to make Promise less error prone to compartment mismatches, r=bzbarsky
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D38762
# ==============================================================================

diff -r c4980c083852 -r ac548c2a725b dom/base/BodyConsumer.cpp
--- a/dom/base/BodyConsumer.cpp	Mon Jul 29 08:00:35 2019 +0000
+++ b/dom/base/BodyConsumer.cpp	Mon Jul 29 14:43:54 2019 +0000
@@ -691,7 +691,7 @@
         JS::Rooted<JS::Value> val(cx);
         val.setObjectOrNull(arrayBuffer);
 
-        localPromise->MaybeResolve(cx, val);
+        localPromise->MaybeResolve(val);
         // ArrayBuffer takes over ownership.
         aResult = nullptr;
       }
@@ -725,7 +725,7 @@
           JS::Rooted<JS::Value> json(cx);
           BodyUtil::ConsumeJson(cx, &json, decoded, error);
           if (!error.Failed()) {
-            localPromise->MaybeResolve(cx, json);
+            localPromise->MaybeResolve(json);
           }
         }
       };